src/components/scenes/Staking/EarnScene.tsx

```diff
@@ -362,7 +366,11 @@ export const EarnScene = (props: Props) => {
             currencyConfig={currencyConfigMap}
             stakePolicy={stakePolicy}
             isOpenPosition={false}
-            onPress={handlePress}
+            onPress={() => {
+              handlePress().catch((error: unknown) => {
+                showError(error)
+              })
+            }}
```

Comment:
Instead of changing this, go to `EarnOptionCard` and just change `onPress: () => Promise<void> | void`. The underlying `EdgeCard` component already understands promises, so it's just a type problem.